### PR TITLE
Feature/drive details api

### PR DESCRIPTION
**Pull request by Sumit-Dhandale**
Drive details API

### CODE REVIEW

File: app/controllers/api/v1/drives_controller.rb [hunk 1]
```diff
@@ -3,8 +3,16 @@
...
-      before_action :set_time_data
+      before_action :load_drive, only: :drive_time_left
+      before_action :set_time_data, only: :drive_time_left
+
+      def show
+        drives_candidate = DrivesCandidate.find_by(token: params[:id])
+        return render_error(message: I18n.t('token_not_found.message'), status: :not_found) unless drives_candidate
+
+        drive = Drive.find(drives_candidate.drive_id)
+        render_success(data: drive, message: I18n.t('ok.message'))
```
Comment (rahulknojha) [inline app/controllers/api/v1/drives_controller.rb:14]: we are not using serializer here ?
For the sake consistency  can we follow same pattern ?

File: app/controllers/api/v1/drives_controller.rb [hunk 2]
```diff
@@ -3,8 +3,16 @@
...
   module V1
     class DrivesController < ApiController
-      before_action :load_drive
-      before_action :set_time_data
+      before_action :load_drive, only: :drive_time_left
+      before_action :set_time_data, only: :drive_time_left
+
+      def show
+        drives_candidate = DrivesCandidate.find_by(token: params[:id])
+        return render_error(message: I18n.t('token_not_found.message'), status: :not_found) unless drives_candidate
```
Comment (rahulknojha) [inline app/controllers/api/v1/drives_controller.rb:11]: can we simply use find, so the the global exception handler will handle this by default

File: spec/controllers/api/v1/drives_controller_spec.rb [hunk 3]
```diff
@@ -3,59 +3,54 @@
...
-        travel_to(DateTime.current.localtime - 1.hours)
-        expect(drive.yet_to_start?).to eq(true)
-      end
-
-      it 'return true if end time > current time' do
-        drive = create(:drive, updated_by_id: @user.id, created_by_id: @user.id,
-                               organization: @organization)
-        travel_to(DateTime.current.localtime - 1.hours)
-        expect(drive.ended?).to eq(false)
+        expect(response.body).to eq('token is not present')
```
Comment (rahulknojha) [inline spec/controllers/api/v1/drives_controller_spec.rb:31]: har coded string, use translation

**Review by rahulknojha — CHANGES_REQUESTED** [review]

File: spec/controllers/api/v1/drives_controller_spec.rb [hunk 4]
```diff
@@ -3,59 +3,54 @@
...
+      it 'fails request' do
+        get :drive_time_left, params: { id: Faker::Number.number }
 
-    context 'ongoing drive' do
-      it 'return true if start time < current time < end time' do
-        drive = create(:drive, updated_by_id: @user.id, created_by_id: @user.id,
-                               organization: @organization)
-        travel 1.hours
-        expect(drive.ongoing?).to eq(true)
+        expect(response.body).to eq('Record not found')
```
Comment (vaibhav8186) [inline spec/controllers/api/v1/drives_controller_spec.rb]: @Sumit-Dhandale   use translation here.

File: app/controllers/api/v1/drives_controller.rb (hunk 1 above)
Comment (somnath77) [reply app/controllers/api/v1/drives_controller.rb]: Used serializer

File: app/controllers/api/v1/drives_controller.rb (hunk 2 above)
Comment (somnath77) [reply app/controllers/api/v1/drives_controller.rb:11]: Finding by token, which is not primary key

File: spec/controllers/api/v1/drives_controller_spec.rb (hunk 3 above)
Comment (somnath77) [reply spec/controllers/api/v1/drives_controller_spec.rb]: used translation

File: app/controllers/api/v1/drives_controller.rb [hunk 5]
```diff
@@ -3,8 +3,16 @@
 module Api
   module V1
     class DrivesController < ApiController
-      before_action :load_drive
-      before_action :set_time_data
+      before_action :load_drive, only: :drive_time_left
+      before_action :set_time_data, only: :drive_time_left
+
+      def show
+        drives_candidate = DrivesCandidate.find_by(token: params[:id])
```
Comment (vaibhav8186) [inline app/controllers/api/v1/drives_controller.rb:10]: @Sumit-Dhandale  we can use active records association here. If we have `drive has many drive_candidates`. Then we can use `drive.drives_candidates.where()`  . It increases the readability of your code as well.

**Review by vaibhav8186 — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/drives_controller.rb (hunk 5 above)
Comment (somnath77) [reply app/controllers/api/v1/drives_controller.rb:10]: It's not possible as we have to find drive details only by using token

File: spec/controllers/api/v1/drives_controller_spec.rb (hunk 4 above)
Comment (somnath77) [reply spec/controllers/api/v1/drives_controller_spec.rb]: Done

**Review by varungadde99 — APPROVED** [review]

**Review by varungadde99 — APPROVED** [review]

File: config/locales/en.yml [hunk 6]
```diff
@@ -41,6 +35,14 @@ en:
     yet_to_start: "Drive is yet to start."
     started: "Drive has already started."
     ended: "Drive has already ended."
+  blank_input:
+    message: "input is empty"
+  ok:
+    message: "ok"
+  token_not_found:
+    message: "token is not present"
```
Comment (vaibhav8186) [inline config/locales/en.yml:43]: @Sumit-Dhandale  Please do proper formating here.` "Token is Not Present"`. Apply this change other place as well.

**Review by vaibhav8186 — CHANGES_REQUESTED** [review]

File: config/locales/en.yml (hunk 6 above)
Comment (somnath77) [reply config/locales/en.yml]: Changes added

File: config/locales/en.yml [hunk 7]
```diff
@@ -41,11 +35,19 @@ en:
...
+  ok:
+    message: "Ok"
+  token_not_found:
+    message: "Token is Not Present"
+  drive_not_found:
+    message: "Drive not found"
   submission:
     not_created:
-      message: "submission not created"
+      message: "Dubmission Not Created"
```
Comment (vaibhav8186) [inline config/locales/en.yml:48]: wrong spelling

**Review by vaibhav8186 — CHANGES_REQUESTED** [review]

**Review by vaibhav8186 — APPROVED** [review]

**Review by rahulknojha — APPROVED** [review]

**Pull request merged**